services/brig/src/Brig/Team/DB.hs

```diff
@@ -174,6 +180,42 @@ lookupInvitationInfo ic@(InvitationCode c)
     cqlInvitationInfo :: PrepQuery R (Identity InvitationCode) (TeamId, InvitationId)
     cqlInvitationInfo = "SELECT team, id FROM team_invitation_info WHERE code = ?"
 
+lookupInvitationByEmail :: (Log.MonadLogger m, MonadClient m) => Email -> m (Maybe Invitation)
+lookupInvitationByEmail e = lookupInvitationEmail e >>= \case
+  Just InvitationInfo {..} -> lookupInvitation iiTeam iiInvId
+  _ -> return Nothing
+
+lookupInvitationEmail :: (Log.MonadLogger m, MonadClient m) => Email -> m (Maybe InvitationInfo)
```

Comment:
```suggestion
lookupInvitationInfoByEmail :: (Log.MonadLogger m, MonadClient m) => Email -> m (Maybe InvitationInfo)
```

Reply:
Also, I'd introduce a custom maybe here with two nothings, and pass that through to the handler where you can map the two different nothings on two different status codes and / or least error labels.

```haskell
data InvitationByEmail =
    InvitationByEmail InvitationInfo
  | InvitationByEmailNotFound
  | InvitationByEmailMoreThanOne
```

Reply:
:+1: